valhalla/thor/dijkstras.h

```diff
@@ -0,0 +1,273 @@
+#ifndef VALHALLA_THOR_Dijkstras_H_
+#define VALHALLA_THOR_Dijkstras_H_
+
+#include <cstdint>
+#include <map>
+#include <memory>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include <valhalla/baldr/double_bucket_queue.h>
+#include <valhalla/baldr/graphid.h>
+#include <valhalla/baldr/graphreader.h>
+#include <valhalla/baldr/location.h>
+#include <valhalla/proto/tripcommon.pb.h>
+#include <valhalla/sif/dynamiccost.h>
+#include <valhalla/sif/edgelabel.h>
+#include <valhalla/thor/edgestatus.h>
+
+namespace valhalla {
+namespace thor {
+
+enum class InfoEdgeType {
+  origin,
+  destination,
+  regular,
+};
+enum class InfoRoutingType {
+  forward,
+  bidirectional,
+  multi_modal,
+};
+
+struct ExpandingNodeMiscInfo {
+  InfoEdgeType edge_type;
+  InfoRoutingType routing_type;
+  bool from_transition;
+};
+
+enum class RouteCallbackRecommendedAction {
+  no_action,
+  skip_expansion,
+  stop_expansion,
+};
+
+/**
+ * Algorithms to do shortest first graph traversal
+ */
+class Dijkstras {
+public:
+  /**
```

Comment:
This doc string isn't super-helpful. Do you think we can drop it altogether? Or add some more info to it, dunno if there's anything to add though. Eg: can the default constructed class be used for routing in any scenario? Or is it not very useful on its own?

Reply:
Yep, good idea. I'll add more here.